Libraries/Interaction/InteractionManager.js

```diff
@@ -73,13 +73,14 @@ var InteractionManager = {
   /**
    * Schedule a function to run after all interactions have completed.
    */
-  runAfterInteractions(callback: Function) {
-    invariant(
-      typeof callback === 'function',
-      'Must specify a function to schedule.'
-    );
-    scheduleUpdate();
-    _queue.push(callback);
+  runAfterInteractions(callback: ?Function): Promise {
+    return new Promise((resolve, reject) => {
+      scheduleUpdate();
+      if (callback) {
+        _queue.push(callback);
+      }
+      _queue.push(resolve);
+    });
```

Comment:
Returning a promise seems fine, but the `callback` has to be executed after interactions, not when the promise is created. How about:

```javascript
runAfterInteractions(callback: ?Function): Promise {
  return new Promise(resolve => {
    scheduleUpdate();
    _queue.push(() => {
      callback && callback();
      resolve();
    });
  });
}
```

Also, both of these implementations will cause the promise to never be resolved if executing `callback` throws an error. Is that intended?